quipucords/api/report_views.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+
+"""Viewset for system report models"""
+import logging
+from django.db.models import Count
+from api.report_model import SystemFingerprint
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+
+# Get an instance of a logger
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class ReportListView(APIView):
+    """List all system reports."""
+
+    def get(self, request):
+        """Lookup and return all system reports."""
+        fact_collection_id = request.query_params.get(
+            'fact_collection_id', None)
+        if fact_collection_id is None:
+            collection_report_list = []
+            # Find all distinct fact_collection_ids
+            fact_collection_value_set = SystemFingerprint.objects.all().values(
+                'fact_collection').distinct()
+
+            # For each id, build a report and add to results array
+            for fact_collection_value in fact_collection_value_set:
+                fact_collection_id = fact_collection_value['fact_collection']
+                report = self.build_report(fact_collection_id)
+                if report is not None:
+                    collection_report_list.append(report)
+                else:
+                    logger.error(
+                        'fact_collection_id %s no longer exists',
```

Comment:
This error really means that no System finger prints could be found associated with the given fact_collection_id (potentially due to some internal delete?).